Replace all API v1 calls in the Admin Panel with Platform API

damianlegawiec opened the pull request:
- [X] Products API
- [X] Classifications/TaxonProducts API
- [X] OptionTypes/OptionValues API
- [X] Users API
- [ ] Variants API
- [x] Taxons API
- [x] Line Items API
- [ ] Shipments API
- [ ] Stock Locations API
- [ ] Stock Movements API
- [ ] Taxonomies API

damianlegawiec committed:
Update Taxon Product management to Platform API

damianlegawiec committed:
Update Taxon Product management to Platform API

damianlegawiec committed:
Use Platform API for Product Picker in Admin Panel

damianlegawiec committed:
Use Platform API for Product Picker in Admin Panel

damianlegawiec committed:
Fixed authorization call in Menu Item Platform API endpoint

damianlegawiec committed:
Fixed authorization call in Menu Item Platform API endpoint

damianlegawiec committed:
Update Platform API Swagger doc

damianlegawiec committed:
Update Platform API Swagger doc

MatthewKennedy reviewed (approved):
All looks good to me so far.

damianlegawiec committed:
Update taxons.js

damianlegawiec committed:
Update taxons.js

damianlegawiec committed:
Merge branch 'feature/platform-api-pt-2' of github.com:spree/spree into feature/platform-api-pt-2

damianlegawiec committed:
Fixed flaky specs plus some CC adjustments

damianlegawiec committed:
Update product_picker.js

damianlegawiec committed:
Use Taxon API v2 for taxon picker

MatthewKennedy commented on backend/app/views/spree/admin/products/_form.html.erb:
```diff
@@ -194,7 +194,14 @@
 
       <% if can? :modify, Spree::Classification %>
-        <%= f.select :taxon_ids, options_from_collection_for_select(@product.taxons, :id, :pretty_name, @product.taxon_ids), { include_hidden: true }, multiple: true, class: 'select2-hidden-accessible' %>
+        <%= f.select :taxon_ids, options_from_collection_for_select(@product.taxons, :id, :pretty_name, @product.taxon_ids), 
```
@damianlegawiec Because The old javascript picks up off the ID here, I was thinking it might be best to change the ID from `product_taxon_ids` to `ProductTaxonIds`

This way we are not running the API V1 and API V2 fetch request together by mistake, and we can ensure that any good results we are getting, are actually coming from the changes we are making, are you okay with this?

MatthewKennedy committed:
Set Variants auto complete to use API V2

MatthewKennedy committed:
Update platform API for variant stock locations and stock items.

MatthewKennedy committed:
Update line_item template

MatthewKennedy committed:
Clean up code for variant_autocomplete.js

MatthewKennedy committed:
Merge branch 'master' into feature/platform-api-pt-2

MatthewKennedy committed:
Move advanceOrder function to APlatform PI V2, convert to ES6 and move it to order.es6 in orders the folder, as it is specific to orders functionality.
Remove duplicate API V1 checkout URL from backend.js
Small Lint fixes, and indentation change to stock transfer file.

MatthewKennedy committed:
Improve orders controller setup - organize backend.js - move old js to orders.es6 for line_items

MatthewKennedy committed:
Fix response in js for advance order, and use lock.

MatthewKennedy committed:
Set Platform API v2 endpoints to build and manage an order.
Set back-end JavaScript to use new endpoints and organize JavaScript.
Add JavaScript functions to show and hide progress indication for use on fetch requests.

MatthewKennedy committed:
Fix test issues.

MatthewKennedy committed:
Add approve to API

MatthewKennedy committed:
update svg

MatthewKennedy committed:
Use flash notice warning of not enough stock.

MatthewKennedy committed:
add shipments_controller Platform API

MatthewKennedy committed:
Organize code for platform API controllers and match services.

MatthewKennedy committed:
Revert change.

MatthewKennedy committed:
Shipment API

MatthewKennedy committed:
Update Shipping Method JS

MatthewKennedy committed:
Strip out as much jQuery as possible from JavaScript pertaining to the admin orders process
Update Order Shipments to use Platform API V2

MatthewKennedy committed:
Update JavaScript

MatthewKennedy committed:
Fix issue with deleting line items from shipment.

MatthewKennedy committed:
Improve JavaScript for variants.

damianlegawiec commented on api/app/controllers/spree/api/v2/platform/shipments_controller.rb:
```diff
@@ -138,7 +138,7 @@ def load_transfer_params
 
           def find_and_update_shipment
-            @shipment = Spree::Shipment.accessible_by(current_ability, :update).readonly(false).find_by!(number: params[:id])
+            @shipment = Spree::Shipment.find_by!(number: params[:id])
```
```suggestion
            @shipment = current_store.shipments..find_by!(number: params[:id])
```

MatthewKennedy committed:
Merge branch 'master' into feature/platform-api-pt-2

MatthewKennedy committed:
Fix promotions error response handling.

MatthewKennedy committed:
Merge branch 'master' into feature/platform-api-pt-2

MatthewKennedy committed:
Move line_items outside order scope, clean up line_items_controller.rb

MatthewKennedy committed:
Replace jQuery with vanilla JS in line_item_managment.js

MatthewKennedy committed:
rename routes for cms_page and cms_section

MatthewKennedy committed:
Clean up controllers and add current_store to finds.

MatthewKennedy committed:
put :cms_page_attribues in alphabetic order.

MatthewKennedy committed:
Push changes to serializers for stock variant and stock location.

MatthewKennedy committed:
Update shipment js

MatthewKennedy committed:
Small JavaScript changes comments and fixes.

MatthewKennedy committed:
Merge branch 'master' into feature/platform-api-pt-2

MatthewKennedy committed:
Add classification_reposition_service

MatthewKennedy committed:
Fix error response handling.

damianlegawiec committed:
Added awesome print to dev/test dependencies

damianlegawiec committed:
Unify pricing formatting by returning decimal numbers

damianlegawiec committed:
Extend Product and Variant serializers to match Storefonts with addional data

damianlegawiec committed:
Merge branch 'master' into feature/platform-api-pt-2
